packages/nimble-components/src/table/specs/table-column-specs/table-column-mapping.md

```diff
@@ -0,0 +1,333 @@
+-   Non-icon, non-spinner Nimble components
+-   Arbitrary html
+
+---
+
+## Design
+
+In addition to the `nimble-table-column-mapping` element, there are `nimble-mapping-<input type>` elements to define mappings between values and html fragments. Below is an example of how these elements would be used within a `nimble-table`:
+
+```HTML
+<nimble-table>
+    <nimble-table-column-mapping field-name="status">
+        Status
+        <nimble-mapping-string key="fail">
+            <template>
+                <nimble-icon-xmark severity="error" title="Failed"></nimble-icon-xmark>
+            </template>
+        </nimble-mapping-string>
+        <nimble-mapping-string key="error">
+            <template>
+                <nimble-icon-xmark severity="error" title="Errored"></nimble-icon-xmark>
+            </template>
+        </nimble-mapping-string>
+        <nimble-mapping-string key="pass">
+            <template>
+                <nimble-icon-check severity="success" title="Passed"></nimble-icon-check>
+            </template>
+        </nimble-mapping-string>
+        <nimble-mapping-string key="running">
+            <template>
+                <nimble-spinner title="Running"></nimble-spinner>
+            </template>
+        </nimble-mapping-string>
+    </nimble-table-column-mapping>
+    <nimble-table-column-mapping field-name="required">
+        Required?
+        <nimble-mapping-boolean key="true">
+            <template>
+                <nimble-icon-check title="Required"></nimble-icon-check>
+            </template>
+        </nimble-mapping-boolean>
+        <nimble-mapping-boolean key="false">
+            <template>
+            </template>
+        </nimble-mapping-boolean>
+    </nimble-table-column-mapping>
+</nimble-table>
+```
+
+Note that the `key` attribute values are always given as strings. In the case of boolean or number mappings, this value is parsed to the appropriate type, so that it can properly be compared to the values from the row records.
+
+If a template includes text, that text should be within a `span` element, otherwise, when the text is too long for the cell, it it will not be ellipsized and there will be no tooltip to show the full value.
+
+```TS
+import {
+    attr,
+    booleanConverter,
+    nullableNumberConverter
+} from '@microsoft/fast-element';
+
+export class TableColumnNumberMapping extends TableColumnMapping {
+    @attr({ converter: nullableNumberConverter })
+    public override key: number | null = null;
+}
+
+export class TableColumnBooleanMapping extends TableColumnMapping {
+    @attr({ converter: booleanConverter })
+    public override key: boolean | null = null;
+}
+
+```
+
+When none of the given mappings match the record value for a cell, that cell will be empty. Alternatively, we could provide support for a default mapping (i.e. `nimble-mapping-default`) that would be rendered when no other values matched. However this doesn't seem worth the effort/complexity.
+
+**Alternatives:**
```

Comment:
Can you discuss the alternative of not slotting elements but instead just exposing configuration via attributes?

In concert with the above proposal where the type of the element is not part of the name, we can then make these represent the visual:

```
nimble-mapping-text key="blah" text="blah"
nimble-mapping-icon key="blah" icon="nimble-icon-blah" title="blah"
nimble-mapping-text-icon key="blah" icon="nimble-icon-blah" title="blah" text="blah"
nimble-mapping-spinner key="blah"
```

I'd like to see a discussion of the pros and cons. I think in this case a big con is giving too much flexibility in the API via the template elements. It's very verbose for the user and I think we would want a lot of validation to be strict about only providing a single icon or a single span or a single version of both in a specific order. I think any flexibility there is a pain.

Reply:
+1

Reply:
After our group discussion of this, I've updated the spec to use this non-template-based approach.